### PR TITLE
Add the ignore-unknown-functions option

### PR DESCRIPTION
**Pull request by Cyberax**
Right now sqlc only supports a handful of built-in functions. Certain extensions like PostGIS add more than a hundred more functions to operate on geospatial data ( https://postgis.net/docs/PostGIS_Special_Functions_Index.html ).

Rather than trying to add all of the function descriptions to sqlc, add an option to ignore unknown functions.

### CODE REVIEW

File: internal/dinosql/query_test.go [hunk 1]
```diff
@@ -927,6 +927,22 @@ func TestComparisonOperators(t *testing.T) {
 	}
 }
 
+func TestUnknownFunctions(t *testing.T) {
+	stmt :=  `
+		CREATE TABLE foo (id text not null);
+		-- name: ListFoos :one
+		SELECT id FROM foo WHERE id = frobnicate($1);
```
Comment (kyleconroy) [inline internal/dinosql/query_test.go]: Excellent fake function name 😆

File: go.sum [hunk 2]
```diff
@@ -10,11 +10,14 @@ github.com/fsnotify/fsnotify v1.4.7/go.mod h1:jwhsz4b93w/PPRr/qN1Yymfu8t87LnFCMo
...
 github.com/google/go-cmp v0.3.0/go.mod h1:8QqcDgzrUqlUb/G2PQTWiueGozuR1884gddMywk6iLU=
 github.com/hashicorp/hcl v1.0.0/go.mod h1:E5yfLk+7swimpb2L/Alb/PJmXilQ/rhwaUYs4T20WEQ=
+github.com/inconshreveable/mousetrap v1.0.0 h1:Z8tu5sraLXCXIcARxBp/8cbvlwVa7Z1NHg9XEKhtSvM=
 github.com/inconshreveable/mousetrap v1.0.0/go.mod h1:PxqpIevigyE2G7u3NXJIT2ANytuPF1OarO4DADm73n8=
 github.com/jinzhu/inflection v1.0.0 h1:K317FqzuhWc8YvSVlFMCCUb36O/S9MCKRDI7QkRKD/E=
 github.com/jinzhu/inflection v1.0.0/go.mod h1:h+uFLlag+Qp1Va5pdKtLDYj+kHp5pxUVkryuEj+Srlc=
+github.com/kr/pretty v0.1.0 h1:L/CwN0zerZDmRFUapSPitk6f+Q3+0za1rQkzVuMiMFI=
 github.com/kr/pretty v0.1.0/go.mod h1:dAy3ld7l9f0ibDNOQOHHMYYIIbhfbHSm3C4ZsoJORNo=
 github.com/kr/pty v1.1.1/go.mod h1:pFQYn66WHrOpPYNljwOMqo10TkYh1fy3cYio2l3bCsQ=
+github.com/kr/text v0.1.0 h1:45sCR5RtlFHMR4UwH9sdQ5TC8v0qDQCHnXt+kaKSTVE=
```
Comment (kyleconroy) [inline go.sum]: Any idea why these three lines were added to `go.sum`? Your changes don't include any new imported packages.

Comment (Cyberax) [reply go.sum]: It looks like it's a duplicate of the line below it without go.mod suffix. I have no idea why Go added it.

Anyway, I removed them from the PR.

**Comment by Cyberax** [comment]
Would you mind merging this PR?

**Comment by kyleconroy** [comment]
Alex, I’m on holiday currently. I’ll be back on Monday, January 6th.

**Comment by Cyberax** [comment]
Sure, I removed the option entirely and made unknown functions work unconditionally.

I'm thinking about a generator that would slurp Postgres function definitions and create a JSON with their description. But that's for a future PR...

**Comment by kyleconroy** [comment]
> I'm thinking about a generator that would slurp Postgres function definitions and create a JSON with their description. But that's for a future PR...

I'm working on a similar change. Adding individual functions by hand isn't scaling.

**Pull request merged**